pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/DataDictionaryExporter.java

```diff
@@ -0,0 +1,293 @@
+            "DATE", "datetime",
+            "DATE_SHORT", "date",
+            "AGREEMENT", "boolean",
+            "BOOLEAN", "boolean",
+            "NUMERIC", "number",
+            "CHECKBOX", "boolean",
+            "NUMBER", "number"
+    );
+
+
+    public String getExportFilename() {
+        return FILE_NAME;
+    }
+
+    /**
+     * initializes the dictionary and internal spreadsheet
+     */
+    public DataDictionaryExporter(List<ModuleExportConfig> configs) {
+        super(configs, null, TabularParticipantExporter.XLSX_FORMAT);
+        wrapStyle = workbook.createCellStyle();
+        wrapStyle.setWrapText(true);
+        boldStyle = workbook.createCellStyle();
+        Font boldFont = workbook.createFont();
+        boldFont.setBold(true);
+        boldStyle.setFont(boldFont);
+
+        boldUnderlineStyle = workbook.createCellStyle();
+        Font boldUnderlineFont = workbook.createFont();
+        boldUnderlineFont.setBold(true);
+        boldUnderlineFont.setUnderline(Font.U_SINGLE);
+        boldUnderlineStyle.setFont(boldUnderlineFont);
+    }
+
+    /** writes the dictionary */
+    public void export(OutputStream os) throws IOException {
+        sheet.setColumnWidth(0, 40 * 256);
+        sheet.setColumnWidth(1, 10 * 256);
+        sheet.setColumnWidth(2, 12 * 256);
+        sheet.setColumnWidth(3, 60 * 256);
+        sheet.setColumnWidth(4, 40 * 256);
+
+        applyToEveryColumn(new DictionaryRowWriter());
+
+        writeAndCloseSheet(os);
+    }
+
+    protected class DictionaryRowWriter implements ColumnProcessor {
+        String previousModuleName = null;
+        String previousQuestionId = null;
+        String previousParentQuestionId = null;
+
+        public void apply(FilterExportConfig filterConfig,
+                          int activityRepeatNum,
+                          int questionRepeatNum,
+                          Map<String, Object> option,
+                          String detailName,
+                          FilterExportConfig parentConfig) {
+            // we don't make separate entries for each iteration of a variable
+            if (activityRepeatNum > 1 || questionRepeatNum > 1) {
+                return;
+            }
+
+            String moduleName = getModuleColumnPrefix(filterConfig, 1);
+            if (!moduleName.equals(previousModuleName)) {
+                addModuleHeaderRows(moduleName, filterConfig.getParent());
+            }
+            String questionId = null;
+            String parentQuestionId = null;
+            if (filterConfig.getQuestionDef() != null) {
+                if (parentConfig != null) {
+                    parentQuestionId = (String) parentConfig.getQuestionDef().get(ESObjectConstants.STABLE_ID);
+                }
+                questionId = (String) filterConfig.getQuestionDef().get(ESObjectConstants.STABLE_ID);
+            }
+
+            if (!StringUtils.equals(previousParentQuestionId, parentQuestionId) && parentQuestionId != null) {
```

Comment:
`isCompositeQustion` could be good name for extracted method for this boolean condition.